server/test/write_authzmodel.go

```diff
@@ -494,38 +796,73 @@ func TestWriteAuthorizationModel(t *testing.T, datastore storage.OpenFGADatastor
 					},
 				},
 			},
-			err: errors.CannotAllowMultipleReferencesToOneRelation,
+			err: errors.NonassignableRelationHasAType("document", "reader"),
 		},
 		{
-			_name: "ExecuteWriteFailsIfIntersectionIncludesSameRelationTwice",
+			name: "user:* is valid when a user type exists",
 			request: &openfgapb.WriteAuthorizationModelRequest{
-				TypeDefinitions: &openfgapb.TypeDefinitions{
-					TypeDefinitions: []*openfgapb.TypeDefinition{
-						{
-							Type: "repo",
-							Relations: map[string]*openfgapb.Userset{
-								"viewer": {
-									Userset: &openfgapb.Userset_Intersection{
-										Intersection: &openfgapb.Usersets{Child: []*openfgapb.Userset{
-											{Userset: &openfgapb.Userset_ComputedUserset{
-												ComputedUserset: &openfgapb.ObjectRelation{
-													Relation: "viewer",
-												},
-											}},
-											{Userset: &openfgapb.Userset_This{}},
-										}},
+				SchemaVersion: "1.1",
+				StoreId:       storeID,
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "document",
+						Relations: map[string]*openfgapb.Userset{
+							"reader": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationalMetadata{
+								"reader": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type:          "user",
+											RelationOrAny: &openfgapb.RelationReference_Any{Any: true},
+										},
 									},
 								},
 							},
 						},
 					},
 				},
 			},
-			err: errors.CannotAllowMultipleReferencesToOneRelation,
+		},
+		{
+			name: "user:* with no user type",
+			request: &openfgapb.WriteAuthorizationModelRequest{
+				SchemaVersion: "1.1",
+				StoreId:       storeID,
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "document",
+						Relations: map[string]*openfgapb.Userset{
+							"reader": {Userset: &openfgapb.Userset_This{}},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationalMetadata{
+								"reader": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{
+											Type:          "user",
+											RelationOrAny: &openfgapb.RelationReference_Any{Any: true},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err: errors.InvalidRelationType("document", "reader", "user", ""),
 		},
 	}
+
+	ctx := context.Background()
+	logger := logger.NewNoopLogger()
+
 	for _, test := range tests {
-		t.Run(test._name, func(t *testing.T) {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
please send a different PR with this change 😬

Reply:
https://github.com/openfga/openfga/pull/238